packages/react-native-sdk/src/providers/StreamCall.tsx

```diff
@@ -134,3 +151,104 @@ const ClearPushWSSubscriptions = () => {
   }, []);
   return null;
 };
+
+/**
+ * This is a renderless component to get the device stats like thermal state and power saver mode.
+ */
+const DeviceStats = () => {
+  const { useCallCallingState } = useCallStateHooks();
+  const callingState = useCallCallingState();
+
+  const [lowPowerMode, setLowPowerMode] = useState<boolean | null>(null);
+  const [thermalState, setThermalState] = useState<string | null>(null);
+
+  const [powerModeSubscription, setPowerModeSubscription] =
+    useState<EmitterSubscription | null>(null);
+  const [thermalStateSubscription, setThermalStateSubscription] =
+    useState<EmitterSubscription | null>(null);
+
+  const { StreamVideoReactNative } = NativeModules;
+  if (!StreamVideoReactNative) {
+    throw new Error('StreamVideoReactNative is not properly linked.');
+  }
```

Comment:
move it out of the render function please, its a global constant

thenafter you can remove it from useEffect dependency